lucene/core/src/java/org/apache/lucene/index/IndexUpgrader.java

```diff
@@ -77,9 +77,10 @@ private static void printUsage() {
   public static void main(String[] args) throws IOException {
     parseArgs(args).upgrade();
   }
-  
+
+  /** Parse arguments. */
   @SuppressForbidden(reason = "System.out required: command line tool")
-  static IndexUpgrader parseArgs(String[] args) throws IOException {
+  public static IndexUpgrader parseArgs(String[] args) throws IOException {
```

Comment:
This have to be public because the renamed o.a.l.backward_index.TestBackwardsCompatibility refers this.

Reply:
Should the test maybe just call `IndexUpgader.main(args)` instead of `upgrader = parseArgs(args) ... upgrader.upgrade()?` @uschindler ?

Reply:
I don't know the code. But maybe indexupgrader can just provide an API to call it without first creating some arguments. This would also make it easier to use for 3rd party code.

Reply:
It should be possible to just call `main()` instead of calling parseArgs() and upgrade() separately. Can we omit the assertion error regarding to `parseArgs()`?

```
        /*
        IndexUpgrader upgrader = null;
        try {
          upgrader = IndexUpgrader.parseArgs(args.toArray(new String[0]));
        } catch (Exception e) {
          throw new AssertionError("unable to parse args: " + args, e);
        }
        upgrader.upgrade();
         */
        // this could throw IOException. 
        // we cannot identify whether the error occurs in parseArgs() or upgrade()
        IndexUpgrader.main(args.toArray(new String[0]));
```

Reply:
I think so.

Reply:
fixed the test and reverted the change for IndexUpgrader. https://github.com/apache/lucene-solr/pull/2000/commits/2b9c1bfbd7ca589a297813938dbbe96472e27ad0